doc/design.md

```diff
@@ -5,55 +5,57 @@ DPPP (DP3, short for Default PreProcessing Pipeline) is a framework
+For each chunk, it calls all the steps to do the processing on this data. The last step will then
+write out the processed data.
 
 ![High level overview of DPPP](doc/images/diagram.png)
 
 DPPP runs a pipeline in clearly defined stages, to make spotting
 errors as easy as possible.
 * **Initialise** This connects all the processing steps together. This step is data independent.
-* **Tune At** this stage only metadata, such as the number of channels, is read, and steps are adjusted to this.
-* **Process** Data is piped through the steps in buffers of time. The buffer size is defined by the steps.
+* **Tune At** This stage reads metadata, such as the number of channels, and adjusts the steps accordingly.
+* **Process** This stage pipes the data through the steps, one buffer at a time. The steps define the buffer size.
 
 ## Measurement Set standard
-DPPP adheres to the Measurement Set (MS) standard, so it
-can be used with data from all radio telescopes. DPPP can
+DPPP adheres to the Measurement Set (MS) standard, which is the common format for radio telescope data. DPPP can
 also create a new MS. When an appropriate input step is in
 place (e.g. a streaming connection to a correlator), MSs can
 be written from scratch.
 
-DPPP supports only regularly shaped MSs, meaning that every
-time slot must have the same number of baselines and antennas.
-Only one spectral window can be used.
+DPPP supports only regularly shaped MSs with the following restrictions:
+- Every time slot has the same number of antennas.
+- Time slots have the same duration and are equally spaced (missing time slots are allowed).
+- Every integration has the same number of correlations and channels.
+- Only one spectral window can be handled.
 
 ## Built-in steps
-DPPP contains built-in steps that are linked together and process one time slot at a time and is therefore suitable for large measurement sets.
-All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class.
+DPPP contains built-in steps that are linked together and process one time slot at a time. This way, DPPP is therefore suitable for large measurement sets.
+All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class. 
+Its inheritance diagram shows all step types, including the built-in steps.
 
 [DPRun](@ref DP3::DPPP::DPRun) initialises and orchestrates the configured steps.
-It begins with setting the required info for all the first step and its next step(s).
+It begins with setting the required info for the first step and its next step(s).
 Next, each buffer (containing a single time slot) is processed by the first step.
 When processed, it invokes the process function of the next step.
-Once all buffers are processed it finishes the processing of the first step and subsequent steps.
-It wraps it with adding some data to the MeasurementSet written/updated which is started at the last step.
+DPPP calls the `finish` function of the first step and subsequent steps. 
```

Comment:
Please restore the 'Once all buffers are processed, ...'. In my original comment I didn't mean to remove it, but I can imagine that wasn't clear.

Reply:
I think that your suggestion makes that sentence unnecessary which is why I deleted it:
_Once all buffers are processed it finishes the processing of the first step and subsequent steps. DPPP calls the `finish` function of the first step and subsequent steps._

Reply:
That's exactly what I meant!